docs/root/start/sandboxes/kafka.rst

```diff
@@ -0,0 +1,127 @@
+.. _install_sandboxes_kafka:
+
+Kafka Filter
+============
+
+In this example, we show how the :ref:`Kafka filter <config_network_filters_kafka_broker>` can be used with the Envoy proxy. The Envoy proxy configuration includes a Kafka filter that undewrstand Kafka network protocol and collects Kafka-specific
+metrics.
+
+Running the Sandboxes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following documentation runs through the setup of both services.
+
+**Step 1: Install Docker**
+
+Ensure that you have a recent versions of ``docker`` and ``docker-compose``.
+
+A simple way to achieve this is via the `Docker Desktop <https://www.docker.com/products/docker-desktop>`_.
+
+**Step 2: Clone the Envoy repo and start all of our containers**
+
+If you have not cloned the Envoy repo, clone it with ``git clone git@github.com:envoyproxy/envoy``
+or ``git clone https://github.com/envoyproxy/envoy.git``
+
+
+Terminal 1
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/kafka
+  $ docker-compose pull
+  $ docker-compose up --build -d
+  $ docker-compose ps
+
+         Name                     Command            State                              Ports
+  ----------------------------------------------------------------------------------------------------------------------
+  kafka_kafka_1       start-kafka.sh                 Up      0.0.0.0:31001->31001/tcp, 9092/tcp
+  kafka_proxy_1       /docker-entrypoint.sh /bin ... Up      10000/tcp, 0.0.0.0:19092->19092/tcp, 0.0.0.0:8001->8001/tcp
+  kafka_zookeeper_1   /bin/sh -c /usr/sbin/sshd  ... Up      2181/tcp, 22/tcp, 2888/tcp, 3888/tcp
+
+
+**Step 3: Produce events**
+
+Use ``kafka scripts`` to issue some commands and verify they are routed via Envoy. Note
+that the current implementation of the protocol filter was tested with Kafka- 2.0. It may, however, not work with other versions of Kafka due to differences
+in the protocol implementation.
+
+Terminal 1
+
+.. code-block:: console
+
+  $ docker run --rm -it --network envoymesh wurstmeister/kafka:2.11-2.0.0 /opt/kafka/bin/kafka-console-producer.sh --broker-list envoy:19092 --topic test_topic
```

Comment:
@adamkotwasinski is about to land support for Kafka 2.4.0 in the Kafka filter (current supported version in master is 2.3.1 - https://github.com/envoyproxy/envoy/pull/9582). The Docker image versions you have here is 2.11 Scala / 2.0.0 Kafka which are about a year old. Should we setup the sandbox using the latest Kafka 2.4.0 container at wurstmeister/kafka:2.12-2.4.0?

Reply:
2.3.1 would be a safer bet, 2.4.0 is going to involve more C++ work than usual

Reply:
Agree. I just tested the sandbox with 2.3.1 and it works great. Have updated my comment.

Reply:
Thanks, I have the change locally. Will push after this CI run finishes just to make sure there are no issues.